lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -715,4 +720,14 @@ private boolean isSchemaOwner(ConnectorSecurityContext context, String schemaNam
         }
         return false;
     }
+
+    private boolean checkFunctionPermission(ConnectorSecurityContext context, FunctionKind functionKind, SchemaRoutineName functionName, Function<FunctionAccessControlRule, Boolean> executePredicate)
```

Comment:
nice! Can you do the same for  FileBasedSystemAccessControl with optional FunctionKind?

Reply:
I would not complicate this